### PR TITLE
[ASCII-145] Extract render subpackge from status

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

The PR uses the work from this [PR](https://github.com/DataDog/datadog-agent/pull/20718) as the base. 

The main takeaways from this PR are:
- Extracted the rendering functionality to a sub-package with the `pkg/status` package. `pkg/status/render` (Probably the name could be change to `pkg/status/format`
- On the package that was importing the entire status package only for the formatting functions, we no longer have to do it. They can import the `status/render` package. The result is less packages are imported
- The last commit modified the `check <check name>` subcommand, so it no longer import the entire `status` package, but rather just `status/collector` and `status/render` 🎉 

### Describe how to test/QA your changes

The output of the status commands works as expected.

Normal agent

- `agent status`
- `agent check [checkName]`
- `agent launch-gui`
- `agent flare [caseID]`

Cluster agent 
- `agent status`
- `agent metamap [nodeName]`
- `agent flare [caseID]`

Process agent 
- `/opt/datadog-agent/embedded/bin/process-agent status`

Security Agent 
- `agent status`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/status/render/helpers.go [hunk 1]
```diff
@@ -72,6 +72,8 @@ func Textfmap() ttemplate.FuncMap {
 	}
 }
 
+var timeFormat = "2006-01-02 15:04:05.999 MST"
```
Comment (dustmop) [inline pkg/status/render/helpers.go:75]:
```suggestion
const timeFormat = "2006-01-02 15:04:05.999 MST"
```

**Review by dustmop — APPROVED** [review]
LGTM, with one suggestion

**Review by dianashevchenko — APPROVED** [review]
Approved, reviewed on the basis of apm files

**Review by paulcacheux — APPROVED** [review]

**Review by sblumenthal — APPROVED** [review]
LGTM for container-integrations files

**Review by kkhor-datadog — COMMENTED** [review]
The changes look fine, but mind updating the QA instructions for the process agent status check? The command should be `/opt/datadog-agent/embedded/bin/process-agent status` because it's a different binary

**Review by kkhor-datadog — APPROVED** [review]

File: pkg/cli/subcommands/check/command.go [hunk 2]
```diff
@@ -363,6 +365,8 @@ func run(config config.Component, cliParams *cliParams, demultiplexer demultiple
 	var checkFileOutput bytes.Buffer
 	var instancesData []interface{}
 	printer := aggregator.AgentDemultiplexerPrinter{DemultiplexerWithAggregator: demultiplexer}
+	collectorData := statuscollector.GetStatus()
+	checkRuns := collectorData["runnerStats"].(map[string]interface{})["Checks"].(map[string]interface{})
```
Comment (hush-hush) [inline pkg/cli/subcommands/check/command.go:369]: I don't understand why we move this variable outside of the for. What is the purpose of this ?

Comment (GustavoCaso) [reply pkg/cli/subcommands/check/command.go:369]: Previously for each check inside the for loop, we were calling `status.GetCheckStatusJSON(c, s)`

```go
// GetCheckStatusJSON gets the status of a single check as JSON
func GetCheckStatusJSON(c check.Check, cs *checkstats.Stats) ([]byte, error) {
	s := collector.GetStatus()

	checks := s["runnerStats"].(map[string]interface{})["Checks"].(map[string]interface{})
	checks[c.String()] = make(map[checkid.ID]interface{})
	checks[c.String()].(map[checkid.ID]interface{})[c.ID()] = cs

	statusJSON, err := json.Marshal(s)
	if err != nil {
		return nil, err
	}

	return statusJSON, nil
}

// GetCheckStatus gets the status of a single check as human-readable text
func GetCheckStatus(c check.Check, cs *checkstats.Stats) ([]byte, error) {
	statusJSON, err := GetCheckStatusJSON(c, cs)
	if err != nil {
		return nil, err
	}

	st, err := renderCheckStats(statusJSON, c.String())
	if err != nil {
		return nil, err
	}
	return []byte(st), nil
}
```

So technically, we collect the collector status information multiple times if there are multiple checks. 

These changes intend to collect the collector status information once, iterate over the check and populate the information. It mostly avoiding fetching the collector status information multiple times. Or at least that is my intention. Did I miss anything @hush-hush ?

**Review by hush-hush — APPROVED** [review]

**Pull request merged**